var/spack/repos/builtin/packages/py-cupy/package.py

```diff
@@ -28,12 +31,16 @@ class PyCupy(PythonPackage, CudaPackage, ROCmPackage):
 
     variant("all", default=False, description="Enable optional py-scipy, optuna, and cython")
 
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
 
     depends_on("python@3.7:", when="@:11", type=("build", "run"))
     depends_on("python@3.8:", when="@12:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-cython@0.29.22:2", type="build")
+    depends_on("py-cython@0.29.22:2", type="build", when="@:13.3.0")
+    depends_on(
+        "py-cython@3:3.0.10,3.0.12:", type="build", when="@13.4.0:"
+    )  # 3.0.11 broken likely because of cython#6335, fixed in 3.0.12
```

Comment:
Possibly, just in case of patch releases, but likely makes no difference:
```suggestion
    depends_on("py-cython@0.29.22:2", type="build", when="@:13.3")
    depends_on(
        "py-cython@3:3.0.10,3.0.12:", type="build", when="@13.4:"
    )  # 3.0.11 broken likely because of cython#6335, fixed in 3.0.12
```